src/attributes/attr.js

```diff
@@ -23,11 +23,11 @@ jQuery.fn.extend({
 
 jQuery.extend({
 	attr: function( elem, name, value ) {
-		var hooks, ret,
+		var ret, hooks, notxml,
 			nType = elem.nodeType;
 
-		// don't get/set attributes on text, comment and attribute nodes
-		if ( nType === 3 || nType === 8 || nType === 2 ) {
+		// Don't get/set attributes on text, comment and attribute nodes
+		if ( !elem || nType === 3 || nType === 8 || nType === 2 ) {
```

Comment:
The lack of this guard in `jQuery.attr` is a behavior difference with respect to `jQuery.prop` since 55ac56aeda786a4d1b677aefd3f5bb134ecb02ad, and if anything the latter should be brought in line (since we read `elem.nodeType` prior to it, we've already thrown if `elem` is `null` or `undefined`). See also #2320.